src/Engine/UnitActionLifecycles/ResupplyLifecycle.java

```diff
@@ -118,37 +104,29 @@ public GameEventQueue getEvents(MapMaster map)
       // Validate action.
       boolean isValid = true;
       isValid &= unitActor != null && !unitActor.isTurnOver;
-      // Unit can move between executions of this action, so verify it's still on the map.
+      isValid &= null != movePath;
       isValid &= (null != map) && map.isLocationValid(unitActor.x, unitActor.y);
+
       if( isValid )
       {
         // Figure out where we are acting.
         supplyLocation = myLocation();
 
-        // Add a move event if we need to move.
-        // Note that movePath being null is OK for ResupplyAction when it is being re-used.
-        if( movePath != null )
+        // If we get blocked, don't resupply anything.
+        if( Utils.enqueueMoveEvent(map, unitActor, movePath, eventSequence) )
         {
-          // If we should be blocked, don't resupply anything.
-          if( !Utils.enqueueMoveEvent(map, unitActor, movePath, eventSequence) )
```

Comment:
That was backwards this whole time, wasn't it?
I cri.

Reply:
It was negated the whole time, but the logic was structured differently so it worked.